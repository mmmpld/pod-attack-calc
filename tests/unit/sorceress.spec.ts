import { describe, it } from 'vitest'
import breakpoints from './breakpoints.js'
const characterId = 6

describe('Sorceress default values', () => {
  it('standard attack', async () => {
    await breakpoints('14', '1.78', { charactersSelected: characterId })
  })

  it('standard attack with giant thresher', async () => {
    await breakpoints('14', '1.78', { charactersSelected: characterId, weaponsPrimarySelectedRaw: 107 })
  })

  it('multishot no-upped doomslinger', async () => {
    await breakpoints('14', '1.78', { charactersSelected: characterId, weaponsPrimarySelectedRaw: 202 })
  })

  it('multishot upped doomslinger no ias', async () => {
    await breakpoints('12', '2.08', { charactersSelected: characterId, weaponsPrimarySelectedRaw: 46 })
  })

  it('multishot upped doomslinger', async () => {
    await breakpoints('11', '2.27', { charactersSelected: characterId, weaponsPrimarySelectedRaw: 46, iasOffWeaponRaw: 10 })
  })
})
