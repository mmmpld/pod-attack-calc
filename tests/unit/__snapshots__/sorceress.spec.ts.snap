// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sorceress default values > multishot no-upped doomslinger 1`] = `
"<tbody>
  <tr class=\\"highlight-current-row\\">
    <th>0</th>
    <td>14</td>
    <td>1.78</td>
  </tr>
  <tr class=\\"\\">
    <th>4</th>
    <td>13</td>
    <td>1.92</td>
  </tr>
  <tr class=\\"\\">
    <th>16</th>
    <td>12</td>
    <td>2.08</td>
  </tr>
  <tr class=\\"\\">
    <th>35</th>
    <td>11</td>
    <td>2.27</td>
  </tr>
</tbody>"
`;

exports[`Sorceress default values > multishot upped doomslinger 1`] = `
"<tbody>
  <tr class=\\"\\">
    <th>0</th>
    <td>12</td>
    <td>2.08</td>
  </tr>
  <tr class=\\"highlight-current-row\\">
    <th>8</th>
    <td>11</td>
    <td>2.27</td>
  </tr>
</tbody>"
`;

exports[`Sorceress default values > multishot upped doomslinger no ias 1`] = `
"<tbody>
  <tr class=\\"highlight-current-row\\">
    <th>0</th>
    <td>12</td>
    <td>2.08</td>
  </tr>
  <tr class=\\"\\">
    <th>8</th>
    <td>11</td>
    <td>2.27</td>
  </tr>
</tbody>"
`;

exports[`Sorceress default values > standard attack 1`] = `
"<tbody>
  <tr class=\\"highlight-current-row\\">
    <th>0</th>
    <td>14</td>
    <td>1.78</td>
  </tr>
  <tr class=\\"\\">
    <th>9</th>
    <td>13</td>
    <td>1.92</td>
  </tr>
  <tr class=\\"\\">
    <th>19</th>
    <td>12</td>
    <td>2.08</td>
  </tr>
  <tr class=\\"\\">
    <th>32</th>
    <td>11</td>
    <td>2.27</td>
  </tr>
  <tr class=\\"\\">
    <th>54</th>
    <td>10</td>
    <td>2.5</td>
  </tr>
  <tr class=\\"\\">
    <th>86</th>
    <td>9</td>
    <td>2.77</td>
  </tr>
  <tr class=\\"\\">
    <th>152</th>
    <td>8</td>
    <td>3.12</td>
  </tr>
</tbody>"
`;

exports[`Sorceress default values > standard attack with giant thresher 1`] = `
"<tbody>
  <tr class=\\"highlight-current-row\\">
    <th>0</th>
    <td>14</td>
    <td>1.78</td>
  </tr>
  <tr class=\\"\\">
    <th>6</th>
    <td>13</td>
    <td>1.92</td>
  </tr>
  <tr class=\\"\\">
    <th>16</th>
    <td>12</td>
    <td>2.08</td>
  </tr>
  <tr class=\\"\\">
    <th>30</th>
    <td>11</td>
    <td>2.27</td>
  </tr>
  <tr class=\\"\\">
    <th>52</th>
    <td>10</td>
    <td>2.5</td>
  </tr>
  <tr class=\\"\\">
    <th>89</th>
    <td>9</td>
    <td>2.77</td>
  </tr>
</tbody>"
`;
